### DISCUSSION TITLE
Add a simple API for overriding/replacing default panels with addons

### DISCUSSION THREAD

**Discussion by eon-s** [godotengine/godot-proposals#5336] [Editor]
Allowing to override or replace panels with an easy to use API, could help on creating addons that turns the editor into an editor for specific types of games, focus the editor on a single subject for educational purposes, offer "beginner" or "simplified" versions of the normal panels as well as advanced versions.

For example, the Tileset editor on Godot 4 is a monster, UX is not the best (it may be improved over time) and is too complex for basic Tileset editing, so if an addon can replace it with a minimal version it may help a lot the most common workflows.

Or replace the tree, you want to add more functionality, separate it in sections, or show the tree as a graph with more details, things like that can help level and UI designers.

**Comment by RedMser** [comment]
You can already do this to some extent by using [feature profiles](https://docs.godotengine.org/en/stable/classes/class_editorfeatureprofile.html) to hide the existing editors, using the editor plugin api to add new ones in their place, then recreating the original editor's functionality.

Alternatively, directly modifying a dock's node setup could also work, although it's harder to make easily toggle-able.

Not sure if a dedicated api for this is required.
